File: RetailCore.Shared/Entities/Customer.cs
using System;

namespace RetailCore.Shared.Entities;

public class Customer: BaseEntity
{
    public Guid? UserId { get; set; } // AspNetUsers (Week 3)
    public string FullName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string? Phone { get; set; }
    public string? Address { get; set; }
    public string? City { get; set; }
    public bool IsActive { get; set; } = true;
}

File: retail-admin-dashboard/src/types/customer.ts
export interface Customer {
  id: string;          
  userId?: string;     
  fullName: string;
  email: string;
  password?: string;
  phone?: string;
  address?: string;
  city?: string;
  isActive: boolean;
  createdDate: string;
}

export interface CreateCustomerRequest {
  fullName: string;
  email: string;
  password?: string;
}

export interface UpdateCustomerRequest {
  fullName: string;
  email: string;
  phone?: string;
  address?: string;
  city?: string;
  isActive: boolean;
}